ubyssey/views.py

```diff
@@ -217,6 +216,58 @@ def author_articles(self, request, slug=None):
 
         return render(request, 'author/articles.html', context)
 
+    def archive(self, request):
+            
+        current_year = datetime.today().year
+        years = []
+        year_query = """SELECT YEAR(published_at) AS year_published,
+                id FROM content_article GROUP BY YEAR(published_at)"""
+        for year in Article.objects.raw(year_query):
+            if(year.year_published is not None):
+                years.append(year.year_published)
+            
+        sections = Section.objects.all()
+            
+        context = {
+            'sections': sections,
+            'years': years
+        }
+            
+        query = request.GET.get('q', None)
+        section_id = request.GET.get('section_id', None)
+        
+        year = request.GET.get('year', None)
+        
+        article_list = Article.objects.filter(is_published=True)
+        if query == "":
+            query = None
+        if year is not None:
+            context['year'] = year
+            article_list = article_list.filter(published_at__icontains=str(year))
+        if query is not None:
+            article_list = article_list.filter(headline__icontains=query)
+            context['q'] = query
+         
+        if section_id is not None:
+            article_list = article_list.filter(section = section_id)
+            context['section_id'] = section_id
+            
+        paginator = Paginator(article_list, 15) # Show 15 articles per page
+        page = request.GET.get('page')
+        try:
+            articles = paginator.page(page)
+        except PageNotAnInteger:
+               # If page is not an integer, deliver first page.
+            articles = paginator.page(1)
+        except EmptyPage:
+            # If page is out of range (e.g. 9999), deliver last page of results.
+            articles = paginator.page(paginator.num_pages)
+            
```

Comment:
`context` should include a `meta` property that defines the page title, something like this:

```python
context['meta'] = { 'title': 'Archive' }
```

Here's an example:
 https://github.com/ubyssey/ubyssey-dispatch-theme/blob/staging/ubyssey/views.py#L286-L288